### DISCUSSION TITLE
How to fetch data before rendering the app without breaking SSG

### DISCUSSION THREAD

**Discussion by yonimor** [vercel/next.js#19831] [Help]
I'm new to next.js and my use case is as follows: I want to deploy a static site to the Vercel CDN, but I also need to make use of request headers before rendering the page.

More specifically - I need to use IP data to block some regions from accessing the site and make privacy adjustments based on local laws.

Technically speaking that means a server side function which runs before loading _document.js.

How is that to be done with next.js?

Thanks in advance!

**Comment by jamesmosier** [comment]
There is no way to do this within Next & Vercel without using `getServerSideProps` since each user that visits your site you will need to check their IP/location. 

The alternative would be setting up an intermediary server that you proxy your traffic to, run the server logic there, and then forward the user to your application if they are allowed to view it. But this just ends up with the same issue of having to run server code before a user gets to your site.

**Reply by yonimor** [reply]
Thanks for the response.

I remember reading somewhere that you can run lambda functions on the edge but I didn't find out how.
No problem running server side code, but it has to happen on the edge.
